contracts/RMRK/extension/multiAssetAutoIndex/RMRKMultiAssetAutoIndex.sol

```diff
@@ -0,0 +1,108 @@
+// SPDX-License-Identifier: Apache-2.0
+
+pragma solidity ^0.8.16;
+
+import "./IRMRKMultiAssetAutoIndex.sol";
+import "../../multiasset/RMRKMultiAsset.sol";
+
+contract RMRKMultiAssetAutoIndex is IRMRKMultiAssetAutoIndex, RMRKMultiAsset {
+    // Mapping of tokenId to assetId to index on the _pendingAssetIndex array
+    mapping(uint256 => mapping(uint256 => uint256)) private _pendingAssetIndex;
+
+    constructor(
+        string memory name_,
+        string memory symbol_
+    ) RMRKMultiAsset(name_, symbol_) {}
+
+    function supportsInterface(
+        bytes4 interfaceId
+    ) public view virtual override(IERC165, RMRKMultiAsset) returns (bool) {
+        return
+            interfaceId == type(IRMRKMultiAssetAutoIndex).interfaceId ||
+            super.supportsInterface(interfaceId);
+    }
+
+    /**
+     * @inheritdoc IRMRKMultiAssetAutoIndex
+     */
+    function acceptAsset(uint256 tokenId, uint64 assetId) public {
+        uint256 index = _pendingAssetIndex[tokenId][assetId];
+        _acceptAsset(tokenId, index, assetId);
+    }
+
+    /**
+     * @inheritdoc IRMRKMultiAssetAutoIndex
+     */
+    function rejectAsset(uint256 tokenId, uint64 assetId) public {
+        uint256 index = _pendingAssetIndex[tokenId][assetId];
+        _rejectAsset(tokenId, index, assetId);
+    }
+
+    /**
+     * @inheritdoc AbstractMultiAsset
+     */
+    function _afterAcceptAsset(
+        uint256 tokenId,
+        uint256 index,
+        uint256 assetId
+    ) internal override {
+        require(
+            _pendingAssetIndex[tokenId][assetId] == index,
+            "Trying to delete asset at the wrong index"
+        );
+
+        uint64[] memory pendingAssetsIds = getPendingAssets(tokenId);
```

Comment:
This loads all assets in memory unnecessarily. Please access directly using `_pendingAssets` to get the length, and to find the `replacingAssetId`.